### DISCUSSION TITLE
Ability to remove data loader suffix from the logger

### DISCUSSION THREAD

**Discussion by fmocking** [Lightning-AI/pytorch-lightning#10733] [Idea pool]
Hello! 

I'm not sure about the best place for this rather it should be an issue or discussion. If I'm not missing anything, currently there is no way to remove `DATALOADER_SUFFIX` from the logger without changing the [ResultCollection](https://github.com/PyTorchLightning/pytorch-lightning/blob/adafb24e0df689b5fbcd4f8a100777467c65f20d/pytorch_lightning/trainer/connectors/logger_connector/result.py#L338) class when multiple data loaders are used in validation/testing. The reason for that is [_forked_name](https://github.com/PyTorchLightning/pytorch-lightning/blob/adafb24e0df689b5fbcd4f8a100777467c65f20d/pytorch_lightning/trainer/connectors/logger_connector/result.py#L523) method is called unconditionally and it appends `DATALOADER_SUFFIX` if `result_metric.meta.dataloader_idx` is not none. Although I mapped the names accordingly with their ids when I'm logging this suffix appended no matter what.  It is a minor issue since it might be ignored easily. But I found it harder to track from ids alone. The easiest way I can think is to add an extra parameter to append the `DATALOADER_SUFFIX` that defaults to true. A little trickier but what I believe is a more convenient way can be allowing data loader dictonaries. This way indexes can be somewhat more meaningful. 

If there is any other way I can prevent it from appending the `DATALOADER_SUFFIX` please let me know.

Thanks,

**Comment by rohitgr7** [comment]
if you mean, while logging, then there is an option for it `self.log(..., add_dataloader_idx=False)`.
